pumpkin-lib/src/propagators/cumulative/options.rs

```diff
@@ -0,0 +1,28 @@
+use super::CumulativeExplanationType;
+
+#[derive(Debug, Copy, Clone, Default)]
+pub struct CumulativeOptions {
+    /// Specifies whether it is allowed to create holes in the domain; if this parameter is set to
+    /// false then it will only adjust the bounds when appropriate rather than removing values from
+    /// the domain
+    pub(crate) allow_holes_in_domain: bool,
+    /// The type of explanation which is used by the cumulative to explain propagations and
+    /// conflicts.
+    pub(crate) explanation_type: CumulativeExplanationType,
+    /// Determines whether a sequence of profiles is generated when explaining a propagation.
+    pub(crate) generate_sequence: bool,
+}
+
+impl CumulativeOptions {
+    pub fn new(
```

Comment:
Does not seem necessary, as any combination of values to the fields on the struct is a valid state of the type. Therefore, they can be public